azkaban-web-server/src/web/js/azkaban/view/svg-graph.js

```diff
@@ -439,6 +441,50 @@ azkaban.SvgGraphView = Backbone.View.extend({
     this.graphBounds = bounds;
   },
 
+  expandAllFlows: function (node) {
+    if (node) {
+      // expands all embedded flows inside given node
+      if (node.type == 'flow') {
+        this.expandFlow(node);
+      }
+
+      if (node.nodes && node.nodes.length > 0) {
+        for (var i = 0; i < node.nodes.length; ++i) {
+          this.expandAllFlows(node.nodes[i]);
+        }
+      }
+    } else {
+      // expands all embedded flows in the graph
+      var nodes = this.model.get("data").nodes;
+      for (var i = 0; i < nodes.length; ++i) {
+        this.expandAllFlows(nodes[i]);
+      }
+    }
+  },
+
+  collapseAllFlows: function (node) {
+    if (node) {
+      // collapse all embedded flows inside given node
+
+      // collapse already rendered nodes of type flow
+      if (node.type == 'flow' && node.gNode) {
+        this.collapseFlow(node);
+      }
+
+      if (node.nodes && node.nodes.length > 0) {
+        for (var i = 0; i < node.nodes.length; ++i) {
```

Comment:
Consider also moving this for-loop inside the `if (node.type == 'flow' && node.gNode)` check?